### PR TITLE
[Amazon Bedrock] Add AmazonBedrockGenerator

### PR DESCRIPTION
**Pull request by julian-risch**
Add a new generator component called AmazonBedrockGenerator to integrate with Amazon Bedrock.

Chat generator can be added with a later PR.

Corresponding issue: https://github.com/deepset-ai/haystack-core-integrations/issues/113

I tested it locally with an AWS account that has Bedrock permissions ([not the default](https://docs.aws.amazon.com/bedrock/latest/userguide/security_iam_id-based-policy-examples.html)):
```python
from amazon_bedrock_haystack.generators.amazon_bedrock import AmazonBedrockGenerator
generator = AmazonBedrockGenerator(model_name="anthropic.claude-v2",aws_access_key_id=aws_access_key_id,aws_secret_access_key=aws_secret_access_key,aws_region_name=aws_region_name)
generator.run("Who is the best American actor?")

{'replies': ['There is no definitive "best" American actor, as acting skill and talent are subjective. However, some commonly cited great American actors include Tom Hanks, Denzel Washington, Meryl Streep, Robert De Niro, Al Pacino, Leonardo DiCaprio, Marlon Brando, Jack Nicholson, Daniel Day-Lewis, Paul Newman, and Morgan Freeman. Choosing a single "best" actor comes down to personal preference and criteria.']}
```

### CODE REVIEW

File: integrations/amazon_bedrock/src/amazon_bedrock_haystack/generators/amazon_bedrock.py [hunk 1]
```diff
@@ -0,0 +1,312 @@
...
+        model_name: str,
+        aws_access_key_id: Optional[str] = None,
+        aws_secret_access_key: Optional[str] = None,
+        aws_session_token: Optional[str] = None,
+        aws_region_name: Optional[str] = None,
+        aws_profile_name: Optional[str] = None,
+        max_length: Optional[int] = 100,
+        **kwargs,
+    ):
+        if model_name is None or len(model_name) == 0:
```
Comment (vblagoje) [inline integrations/amazon_bedrock/src/amazon_bedrock_haystack/generators/amazon_bedrock.py:83]: I think we can simply check Truthy/Falsy and raise exception

**Review by vblagoje — COMMENTED** [review]
Do we plan to add support for ChatMessage(s) where applicable?

File: integrations/amazon_bedrock/src/amazon_bedrock_haystack/generators/amazon_bedrock.py (hunk 1 above)
Comment (julian-risch) [reply integrations/amazon_bedrock/src/amazon_bedrock_haystack/generators/amazon_bedrock.py]: Done

**Comment by julian-risch** [comment]
> Do we plan to add support for ChatMessage(s) where applicable?

Not as part of this PR I'd suggest but in a separate one, yes.

**Review by vblagoje — APPROVED** [review]
Approving. We made an agreement (Julian and I) that we should add support for chat models (ChatMessage handling), and that we should add metadata where possible and match the current metadata (which we still need for this generator). The goal is to match the format of other generators as much as possible.

**Comment by lambda-science** [comment]
Hey, commenting here because I used both AWS BedRock on haystack 1.x and 2.x.
On 1.x it was working fine. In my pipeline such as: 
```python
AWS_PROFILE_NAME = os.environ.get("AWS_PROFILE_NAME")
AWS_REGION_NAME = os.environ.get("AWS_REGION_NAME")
prompt_model = PromptModel(model_name_or_path="anthropic.claude-instant-v1", max_length=512, model_kwargs={"temperature": 0, "aws_profile_name": AWS_PROFILE_NAME, "aws_region_name": AWS_REGION_NAME})
```
Now I'm migrating to 2.x and it doesnt work anymore:
In 2.x pipeline:
```python
AWS_PROFILE_NAME = os.environ.get("AWS_PROFILE_NAME")
AWS_REGION_NAME = os.environ.get("AWS_REGION_NAME")
generator = AmazonBedrockGenerator(model_name="anthropic.claude-instant-v1",aws_profile_name=AWS_PROFILE_NAME, aws_region_name=AWS_PROFILE_NAME,  max_length=512, model_kwargs={"temperature": 0})
```

And I get the following error:

```python
File "/usr/local/lib/python3.11/site-packages/haystack/core/pipeline/pipeline.py", line 645, in _run_component
2024-01-09T14:42:00.454020807Z     raise PipelineRuntimeError(
2024-01-09T14:42:00.454109035Z haystack.core.errors.PipelineRuntimeError: Generator raised 'EndpointConnectionError: Could not connect to the endpoint URL: "https://bedrock-runtime.<MY_AWS_PROFILE_NAME>.amazonaws.com/model/anthropic.claude-instant-v1/invoke"' 
2024-01-09T14:42:00.454222360Z Inputs: {'prompt': " <my very long prompt>"}
2024-01-09T14:42:00.455094381Z 
```

**Comment by julian-risch** [comment]
@lambda-science There isn't really much of a difference in the implementation of 1.x and 2.x. I couldn't reproduce your issue and was able to run the following:
```python
...
generator = AmazonBedrockGenerator(model_name="anthropic.claude-instant-v1",aws_access_key_id=aws_access_key_id,aws_secret_access_key=aws_secret_access_key,aws_region_name=aws_region_name,max_length=512, model_kwargs={"temperature": 0})
>>> generator.run("Who is the best American actor?")
{'replies': ['There is no definitive "best" actor, as artistic achievement is subjective. Some actors often cited among the greatest of American cinema include Marlon Brando, Gary Cooper, Henry Fonda, James Dean, Morgan Freeman, Daniel Day-Lewis and Meryl Streep.']}
```
If you think the problem is about the implementation in 2.x, could you please open an issue in this repository? Thank you!

**Comment by lambda-science** [comment]
> ```python
> aws_profile_name=AWS_PROFILE_NAME, aws_region_name=AWS_PROFILE_NAME
> ```

Everything is fine I'm just a stupid boy, puting same variable twice :) Thanks !
`aws_profile_name=AWS_PROFILE_NAME, aws_region_name=AWS_PROFILE_NAME`. Not good region = endpoint unreachable of course !

**Pull request merged**